### ISSUE TITLE
Microsoft.CodeAnalysis.NetAnalyzers warns that I should be using version 7.0.0; but there isn't a release version of 7.0.0 yet

### ISSUE THREAD

**Issue by richardissimo** [dotnet/roslyn#65581]
**Version Used**: 
microsoft.codeanalysis.netanalyzers 6.0.0

**Steps to Reproduce**:

1. Create a .Net SDK project (e.g. C# library, .net 6)
2. Add a package reference to microsoft.codeanalysis.netanalyzers\6.0.0
3. Compile it (after the prerelease versions of that package are published but before the v7.0.0 of that package is released)

**Diagnostic Id**:

`C:\Users\redacted\.nuget\packages\microsoft.codeanalysis.netanalyzers\6.0.0\build\Microsoft.CodeAnalysis.NetAnalyzers.targets(591,5): warning : The .NET SDK has newer analyzers with version '7.0.0' than what version '6.0.0' of 'Microsoft.CodeAnalysis.NetAnalyzers' package provides. Update or remove this package reference.`

**Expected Behavior**:
No warning should be given because there isn't a _release_ version of 7.0.0 yet. I would only expect this complaint to be made when there is a _release_ version available.

**Actual Behavior**:
The warning is given that I'm using version 6.0.0; and that there is a 7.0.0. Except there _isn't_ a release version of 7.0.0 yet - there are currently 13 pre-release versions of 7.0.0; but no release version yet.

**Comment by Youssef1313** [comment]
This belongs to dotnet/roslyn-analyzers, and I think the message is pretty much by-design.

Which version of the .NET SDK are you using? (I see you mentioned you have .NET 6 library, but I assume you're referring to TargetFramework, not SDK version)
If you're using .NET 7 SDK, then this warning is by-design. As the message mentions, you could remove the package reference and get the analyzers from SDK directly. You *could* silence the message by specifying `<_SkipUpgradeNetAnalyzersNuGetWarning`>true</_SkipUpgradeNetAnalyzersNuGetWarning>`

**Comment by richardissimo** [comment]
> That would generally be fixed when 7.0.0 is released, it's just a matter of time.

@Youssef1313 Whilst that is true, it's also just a matter of time before this issue occurs again, with 8.0.0, and then 9.0.0, etc.

**Comment by oskrabanek** [comment]
It's still not fixed. Any progress on this issue?

**Comment by edgarssults** [comment]
I am also having this issue and would like the warning to not take into account preview versions.

We have upgraded to 7.0.0 analyzers recently and just started getting warnings again about upgrading to 7.0.1, but there are only preview versions of it available right now.

**Comment by RobSwDev** [comment]
We're also hitting this issue.

**Comment by bobwah** [comment]
> > That would generally be fixed when 7.0.0 is released, it's just a matter of time.
> 
> @Youssef1313 Whilst that is true, it's also just a matter of time before this issue occurs again, with 8.0.0, and then 9.0.0, etc.

Happening again 7.0.1

**Comment by mavasani** [comment]
I’ll work on a fix here..

**Comment by LukeOrdelmans** [comment]
Adding this (in .csproj / Directory.Build.props) worked for me:

```  
<PropertyGroup>
   <_SkipUpgradeNetAnalyzersNuGetWarning>true</_SkipUpgradeNetAnalyzersNuGetWarning>
</PropertyGroup>
```

**Comment by mavasani** [comment]
7.0.1 is now released on NuGet

**Comment by hoerup** [comment]
And now you have done exactly the same with 7.0.3 :(

**Comment by Pentadome** [comment]
>The .NET SDK has newer analyzers with version '7.0.3' than what version '7.0.1' of 'Microsoft.CodeAnalysis.NetAnalyzers' package provides.

Can confirm that this issue is not fixed yet. [There is no 7.0.3 version of 'Microsoft.CodeAnalysis.NetAnalyzers' on Nuget](https://www.nuget.org/packages/Microsoft.CodeAnalysis.NetAnalyzers/7.0.1?_src=template#versions-body-tab)

**Comment by mavasani** [comment]
https://www.nuget.org/packages/Microsoft.CodeAnalysis.NetAnalyzers/7.0.3 is now available.

I have also started an offline email thread to ensure that we synchronize the publishing of the NuGet package with the .NET SDK releases. We may also consider completely removing this warning. I'll post an update here once we have consensus.

Sorry for the inconvenience and thank you for your patience!

**Comment by b-f-** [comment]
![image](https://github.com/dotnet/roslyn/assets/2358754/a83b1e3d-3a24-43ab-8cbd-8df406b808ff)
Seems like the warning is always a 0.0.1 ahead of actual latest release. Looks like package version is out of sync.

**Comment by hoerup** [comment]
> I have also started an offline email thread to ensure that we synchronize the publishing of the NuGet package with the .NET SDK releases. We may also consider completely removing this warning. I'll post an update here once we have consensus.
> 
@mavasani Since this situation now also occurred with 7.0.4 i assume that the mentioned email thread didn't come up with a solution ?

**Comment by odinserj** [comment]
I have the same warning, and version 7.0.4 is still pre-release only. I don't want to disable this warning entirely, because it can be useful in future. But in my opinion, it should not take into account *preview* versions.

```
Microsoft.CodeAnalysis.NetAnalyzers.targets(593,5): Warning  : The .NET SDK has newer analyzers with version '7.0.4' than what version '7.0.3' of 'Microsoft.CodeAnalysis.NetAnalyzers' package provides. Update or remove this package reference.
```
